File: lecture-recording-manager/Models/Lecture.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Threading.Tasks;

namespace lecture_recording_manager.Models
{
    public class Lecture
    {
        public int Id { get; set; }

        [Required]
        [Column(TypeName = "varchar(255)")]
        public string Title { get; set; }

        [Column(TypeName = "text")]
        public string Description { get; set; }

        [Required]
        public int SemesterId { get; set; }

        public Semester Semester { get; set; }

        public bool Publish { get; set; }

        public bool Active { get; set; }
    }
}

File: lecture-recording-manager/ClientApp/src/app/shared/lecture.ts
import { Semester } from './semester';

export class Lecture {
  public id: number;
  public shortHand: string;
  public title: string;
  public description?: string;

  public semesterId?: number;
  public semester?: Semester;

  public publish: boolean;
  public active: boolean;
  public renderFullHd: boolean;
  public lastSynchronized?: Date;

  public publishPath: string;
  public sourcePath: string;
  public convertedPath: string;
}
